graphql/query_resolver.go

```diff
@@ -338,14 +338,23 @@ func (r *queryResolver) Hosts(ctx context.Context, hostID *string, distroID *str
 		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error getting hosts: %s", err.Error()))
 	}
 
-	apiHosts := []*restModel.APIHost{}
+	usr := mustHaveUser(ctx)
+	forbiddenHosts := []string{}
+	for _, h := range hosts {
+		if !userHasHostPermission(usr, h.Distro.Id, evergreen.HostsView.Value) {
+			forbiddenHosts = append(forbiddenHosts, h.Id)
+		}
+	}
+	if len(forbiddenHosts) > 0 {
+		return nil, Forbidden.Send(ctx, fmt.Sprintf("user '%s' does not have permission to access one or more hosts.", usr.Username()))
```

Comment:
nit. formatting (no period)
```suggestion
		return nil, Forbidden.Send(ctx, fmt.Sprintf("user '%s' does not have permission to access one or more hosts", usr.Username()))
```